hashicorp/web-unified-docs
# feat: optional productSlugs param for all-docs-paths

### Description
This PR adds support for passing products to the `api/all-docs-paths` route. This is necessary so only the paths from product migrated to UDR are used in dev-portal. Now `api/all-docs-paths` will return all the paths, `api/all-docs-paths/terraform` will return only terraform paths, and `api/all-docs-paths/terraform/ptfe-releases/...` will return paths for terraform, ptfe-releases, and any other products after that.

EDIT: also this PR now changes the way that docsPaths data is retrieved. Now there is a script that runs during pre-build that creates a JSON file containing all the paths. The paths are then retrieved from that JSON file when the `/all-docs-paths` endpoint is hit.

### Testing
1. Go to [/api/all-docs-paths](https://web-unified-docs-jucr8feep-hashicorp.vercel.app/api/all-docs-paths)
2. This should show all paths from UDR
3. Go to [/api/all-docs-paths?products=terraform](https://web-unified-docs-jucr8feep-hashicorp.vercel.app/api/all-docs-paths?products=terraform)
4. This should show only paths for the pages in the terraform section
5. Go to [/api/all-docs-paths?products=terraform&products=terraform-docs-common](https://web-unified-docs-jucr8feep-hashicorp.vercel.app/api/all-docs-paths?products=terraform&products=terraform-docs-common)
6. This should show only paths for the pages in the terraform and terraform-docs-common section
